server.js

```diff
@@ -4960,6 +4960,27 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Facebook integration.
+camp.route(/^\/facebook\/share\/([^\/]+)\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var scheme = match[1]; // eg, https
+  var path = match[2];   // eg, shields.io
+  var format = match[3];
+  var page = encodeURIComponent(scheme + '://' + path);
+  var badgeData = getBadgeData('Share', data);
+  if (badgeData.template === 'social') {
+    badgeData.logo = badgeData.logo || logos.facebook;
+    badgeData.links = [
+      'https://www.facebook.com/sharer/sharer.php?u=' + page,
+      'https://www.facebook.com/search?q=' + page,
+     ];
+  }
+  badgeData.text[1] = '';
```

Comment:
That _looks_ ok but I don't understand the mechanics yet. In `suggest.js` there's a `twitterPage` that still produces a badge pointing to `https://img.shields.io/twitter/url/`. That's also what I'm seeing when I put `http://shields.io` into the search bar and press "Suggest badges". I was checking INSTALL.md and CONTRIBUTING.md but I didn't find any mention of `suggest.js`.
Furthermore, I'd like this new Facebook share badge be listed in the "Social" section on http://shields.io/ which won't happen if it's only in `suggest.js`, right?

> add the right-hand-side by performing a call to Facebook's graph API

As suggested above I see that as a separate badge (I won't be implementing that right now).